exporter/chronicleexporter/hostmetrics.go

```diff
@@ -0,0 +1,143 @@
+package chronicleexporter
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/observiq/bindplane-agent/exporter/chronicleexporter/protos/api"
+	"github.com/shirou/gopsutil/v3/process"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type exporterMetrics struct {
+	mutex       sync.Mutex
+	agentID     []byte
+	customerID  []byte
+	exporterID  string
+	namespace   string
+	startTime   *timestamppb.Timestamp
+	stats       *api.AgentStatsEvent
+	logsDropped int64
+	logsSent    int64
+}
+
+func newExporterMetrics(agentID, customerID []byte, exporterID, namespace string) *exporterMetrics {
+	now := timestamppb.Now()
+	return &exporterMetrics{
+		agentID:    agentID,
+		exporterID: exporterID,
+		startTime:  now,
+		customerID: customerID,
+		namespace:  namespace,
+		stats: &api.AgentStatsEvent{
+			WindowStartTime: now,
+			AgentId:         agentID,
+			StartTime:       now,
+		},
+	}
+}
+
+func (cm *exporterMetrics) getAndReset() *api.BatchCreateEventsRequest {
+	cm.mutex.Lock()
+	defer cm.mutex.Unlock()
+
+	batchID := uuid.New()
+	source := &api.EventSource{
+		CollectorId: chronicleCollectorID[:],
+		Namespace:   cm.namespace,
+		CustomerId:  cm.customerID,
+	}
+
+	request := &api.BatchCreateEventsRequest{
+		Batch: &api.EventBatch{
+			Id:        batchID[:],
+			Source:    source,
+			Type:      api.EventBatch_AGENT_STATS,
+			StartTime: cm.startTime,
+			Events: []*api.Event{
+				{
+					Timestamp:      timestamppb.Now(),
+					CollectionTime: timestamppb.Now(),
```

Comment:
Seems like these should be the same timestamp (one call to timestamppb.Now)

Reply:
updated